File: pkg/codegen/testing/test/testdata/implicit-string-range-pp/python/implicit-string-range.py
import pulumi
import pulumi_aws as aws

config = pulumi.Config()
# Number of AZs to cover in a given region
az_count = config.get("azCount")
if az_count is None:
    az_count = "10"
buckets_per_availability_zone = []
for range in [{"value": i} for i in range(0, int(az_count))]:
    buckets_per_availability_zone.append(aws.s3.Bucket(f"bucketsPerAvailabilityZone-{range['value']}", website=aws.s3.BucketWebsiteArgs(
        index_document="index.html",
    )))

File: pkg/codegen/testing/test/testdata/implicit-string-range-pp/dotnet/implicit-string-range.cs
using System.Collections.Generic;
using System.Linq;
using Pulumi;
using Aws = Pulumi.Aws;

return await Deployment.RunAsync(() => 
{
    var config = new Config();
    // Number of AZs to cover in a given region
    var azCount = config.Get("azCount") ?? "10";
    var bucketsPerAvailabilityZone = new List<Aws.S3.Bucket>();
    for (var rangeIndex = 0; rangeIndex < int.Parse(azCount); rangeIndex++)
    {
        var range = new { Value = rangeIndex };
        bucketsPerAvailabilityZone.Add(new Aws.S3.Bucket($"bucketsPerAvailabilityZone-{range.Value}", new()
        {
            Website = new Aws.S3.Inputs.BucketWebsiteArgs
            {
                IndexDocument = "index.html",
            },
        }));
    }
});

